pkg/cli/config.go

```diff
@@ -43,33 +44,41 @@ type Config struct {
 }
 
 func (c *Config) ConnectionControlDirExpanded() string {
-	return expandPath(c.ConnectionControlDir)
+	return ExpandPath(c.ConnectionControlDir)
 }
 
 func (c *Config) LogFilesDeleteThreshold() time.Duration {
 	return time.Duration(c.KeepLogFilesDays*24) * time.Hour
 }
 
-func DefaultConfig() Config {
-	return Config{
+func BaseConfig() *Config {
+	return &Config{
 		ConnectionControlDir: "~/.cvdr/connections",
 		KeepLogFilesDays:     30, // A default is needed to not keep forever
 	}
 }
 
-func ParseConfigFile(config *Config, confFile io.Reader) error {
-	decoder := toml.NewDecoder(confFile)
-	// Fail if there is some unknown configuration. This is better than silently
-	// ignoring a (perhaps mispelled) config entry.
-	decoder.Strict(true)
-	err := decoder.Decode(config)
+func LoadConfigFile(path string) (*Config, error) {
+	b, err := os.ReadFile(path)
 	if err != nil {
-		return err
+		return nil, fmt.Errorf("Error reading config file: %w", err)
 	}
-	return nil
+	return buildConfig([]io.Reader{bytes.NewReader(b)})
 }
 
-func expandPath(path string) string {
+func LoadWithSystemAndUserConfigFiles(system, user string) (*Config, error) {
+	systemb, err := os.ReadFile(system)
+	if err != nil {
+		return nil, fmt.Errorf("Error reading system config file: %w", err)
+	}
+	userb, err := os.ReadFile(user)
+	if err != nil {
+		return nil, fmt.Errorf("Error reading user config file: %w", err)
+	}
+	return buildConfig([]io.Reader{bytes.NewReader(systemb), bytes.NewReader(userb)})
```

Comment:
why not use the file readers in this line too?

Reply:
true, it's no longer relevant though.